protocols/mdns/src/behaviour.rs

```diff
@@ -279,29 +316,57 @@ impl NetworkBehaviour for Mdns {
     > {
         while let Poll::Ready(event) = Pin::new(&mut self.if_watch).poll(cx) {
             let multicast = From::from([224, 0, 0, 251]);
+            let multicast_v6 = Ipv6Addr::new(0xFF02, 0, 0, 0, 0, 0, 0, 0xFB);
             let socket = self.recv_socket.get_ref();
             match event {
                 Ok(IfEvent::Up(inet)) => {
                     if inet.addr().is_loopback() {
                         continue;
                     }
-                    if let IpAddr::V4(addr) = inet.addr() {
-                        log::trace!("joining multicast on iface {}", addr);
-                        if let Err(err) = socket.join_multicast_v4(&multicast, &addr) {
-                            log::error!("join multicast failed: {}", err);
-                        } else {
-                            self.send_buffer.push_back(build_query());
+                    match self.domain {
+                        IPType::IPv4 => {
+                            if let IpAddr::V4(addr) = inet.addr() {
```

Comment:
match on address type instead of self.domain

Reply:
done